src/number/n_transExt.jl

```diff
@@ -0,0 +1,360 @@
+export n_transExt, N_FField, transcendence_degree, transcendence_basis
+
+###############################################################################
+#
+#   Data type and parent methods
+#
+###############################################################################
+
+elem_type(::Type{N_FField}) = n_transExt
+
+parent(a::n_transExt) = a.parent
+
+parent_type(::Type{n_transExt}) = N_FField
+
+base_ring(a::n_transExt) = Union{}
+
+base_ring(a::N_FField) = a.base_ring
+
+@doc Markdown.doc"""
+    transcendence_degree(F::N_FField)
+> Return the trancendence degree of the given function field.
+"""
+transcendence_degree(F::N_FField) = Int(libSingular.rPar(F.ptr))
+
+@doc Markdown.doc"""
+    basis(F::N_FField)
+> Return the trancendence basis of the given function field.
+"""
+function transcendence_basis(F::N_FField)
+   n = transcendence_degree(F)
+   return [F(libSingular.n_Param(Cint(i), F.ptr)) for i = 1:n]
+end
+
+@doc Markdown.doc"""
+   characteristic(R::N_FField)
+> Return the characteristic of the field.
+"""
+function characteristic(R::N_FField)
+   return ZZ(libSingular.n_GetChar(R.ptr))
+end
+
+function deepcopy_internal(a::n_transExt, dict::IdDict)
+   return parent(a)(libSingular.n_Copy(a.ptr, parent(a).ptr))
+end
+
+#TODO: rewrite hash function
+function hash(a::n_transExt, h::UInt)
```

Comment:
We really need a working hash. You can take a look at some of the other hash functions to see how to do it. The main feature of a hash is that all the data should be mixed in somehow. Arithmetically equal objects should have the same hash where possible. Extra points if rearranging the data in the object doesn't lead to the same hash (this can usually be done by rotating the data mod 2^64 by a few bits and xoring).

Reply:
Done